ethcore/src/pod_state.rs

```diff
@@ -65,31 +57,28 @@ impl From<ethjson::spec::State> for PodState {
 	}
 }
 
-impl fmt::Display for PodState {
-	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		for (add, acc) in &self.0 {
-			writeln!(f, "{} => {}", add, acc)?;
-		}
-		Ok(())
+impl From<BTreeMap<Address, PodAccount>> for PodState {
+	fn from(s: BTreeMap<Address, PodAccount>) -> Self {
+		PodState(s)
 	}
 }
 
 /// Calculate and return diff between `pre` state and `post` state.
 pub fn diff_pod(pre: &PodState, post: &PodState) -> StateDiff {
 	StateDiff {
-		raw: pre.get().keys()
-			.merge(post.get().keys())
-			.filter_map(|acc| pod_account::diff_pod(pre.get().get(acc), post.get().get(acc)).map(|d| (acc.clone(), d)))
+		raw: pre.0.keys()
+			.chain(post.0.keys())
```

Comment:
`merge` chains and sorts iterated values, but it is not needed, as we later on, collect all mapped values into `BTreeMap` and sort them again